### PR TITLE
Re send mail issue

### PR DESCRIPTION
**Pull request by ankit213**
I have use exceptions, for check what is throwing the actual error.

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs [hunk 1]
```diff
@@ -7,10 +7,9 @@
 using Microsoft.AspNetCore.Identity;
 using Promact.Oauth.Server.Models.ApplicationClasses;
 using System.Threading.Tasks;
-using System.Linq;
+using Exceptionless;
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs]: I am not sure why have you added Exceptionless here in using. But it serves no purpose. Please remove all unused usings

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs [hunk 2]
```diff
@@ -20,19 +19,18 @@ namespace Promact.Oauth.Server.Controllers
     public class UserController : BaseController
     {
         #region "Private Variable(s)"
-        private readonly ILogger _logger;
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs]: Why have you removed logger? logger is useful for logging messages. Please incorporate logger properly.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs [hunk 3]
```diff
@@ -408,7 +408,8 @@ public bool FindUserBySlackUserName(string slackUserName)
             }
             catch (Exception ex)
             {
-                throw ex;
+                ex.ToExceptionless().Submit();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: No need to call ToExceptionless explicitly just throw the exception as it is.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs [hunk 4]
```diff
@@ -46,18 +45,17 @@ public Task SendEmailAsync(string email, string subject, string message)
             {
                 using (var smtp = new SmtpClient())
                 {
-                    _logger.LogInformation("Start Email Sending");
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: Please don't remove loggers. They are essential

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs [hunk 5]
```diff
@@ -46,18 +45,17 @@ public Task SendEmailAsync(string email, string subject, string message)
...
+                  
                     smtp.ConnectAsync(_appSettings.Value.Host, _appSettings.Value.Port, MailKit.Security.SecureSocketOptions.None).Wait();
                     smtp.AuthenticateAsync(credentials: new NetworkCredential(_appSettings.Value.UserName, _appSettings.Value.Password)).Wait();
-                    smtp.SendAsync(msg, CancellationToken.None).Wait();
-                    smtp.DisconnectAsync(true, CancellationToken.None).Wait();
+                    smtp.SendAsync(msg).Wait();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: Calling an async method and calling wait does not make sense here. either keep Async properly or use synchronous method.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs [hunk 6]
```diff
@@ -46,18 +45,17 @@ public Task SendEmailAsync(string email, string subject, string message)
...
+                    smtp.DisconnectAsync(true).Wait();
                     return Task.FromResult(0);
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: I am not sure why it should return 0. Please correct this.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs [hunk 7]
```diff
@@ -46,18 +45,17 @@ public Task SendEmailAsync(string email, string subject, string message)
...
                 }
             }
             catch (Exception ex)
             {
-                _logger.LogInformation("Throw Email Error", ex);
-                _logger.LogInformation("Throw Email Error Message", ex.Message);
+                ex.ToExceptionless().Submit();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: Don't use Exceptionless explicitly. Loggers are good enough but remove try catch all together for better exception Handling.

Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: but sir, logger are not implemented in OAuth ,
Only exceptionless has been implemented by Siddharth

Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: Yes, Remove all try catch

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs (hunk 1 above)
Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs]: removed exceptionless from UserController.cs

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs (hunk 3 above)
Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs]: Removed All try Catch

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs (hunk 4 above)
Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: Added Logger in Message Service

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs (hunk 5 above)
Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: Now change removed Async method , used the synchronous method
 smtp.Connect(_appSettings.Value.Host,_appSettings.Value.Port,MailKit.Security.SecureSocketOptions.None);
 smtp.Authenticate(credentials: new NetworkCredential(_appSettings.Value.UserName, _appSettings.Value.Password));
 smtp.Send(msg, CancellationToken.None);
 smtp.Disconnect(true, CancellationToken.None);

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs (hunk 6 above)
Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Services/MessageServices.cs]: I am also not sure about why use return 0, So i change send mail method, right now it's void method so no need to return.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs (hunk 2 above)
Comment (ankit213) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs]: I have added logger in User Controller

**Pull request merged**